definitions/push-application-snapshot/push-application-snapshot.yaml

```diff
@@ -0,0 +1,49 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: push-application-snapshot-task
+spec:
+  params:
+    - name: mappedApplicationSnapshot
+      description: JSON string representing the ApplicationSnapshot
+      type: string
+    - name: retries
+      description: Retry copy N times.
+      type: int
+      default: 0
+  steps:
+    - name: skopeo-copy
+      onError: stopAndFail
+      image: http://quay.io/hacbs-release/release-utils:latest
+      script: |
+        #!/usr/bin/env bash
+        set -eu
+        app=$(jq -r '.application' <<<$(params.mappedApplicationSnapshot))
+        printf 'Begining "%s" for "%s"\n\n' "$(context.task.name)" "$app"
+        while read line;
+        do
+          set -- $(jq -cr 'to_entries[]| [.key, .value]| @tsv' <<<"$line" | xargs)
+          declare -A map
+          while [[ $# != 0 ]]
+          do  map["$1"]="$2"
+              shift 2
+          done
+          s_Digest=$(skopeo inspect --format '{{.Digest}}' "${map[pullspec]}" 2>/dev/null)
+          d_Digest=$(skopeo inspect --format '{{.Digest}}' "${map[repository]}" 2>/dev/null)
+          if [[ "$d_Digest" != "$s_Digest" || -z "$d_Digest" ]]
+          then
+            printf '* Processing component: %s\n' "${map[component]}"
+            skopeo copy \
+              --all \
+              --preserve-digests \
+              --dest-precompute-digests \
+              --retry-times="$(params.retries)" \
+              "docker://${map[pullspec]}" \
+              "docker://${map[repository]}${map[tag]:+:${map[tag]}}" ;
+          else
+            printf '* Skipping component: %s\n' "${map[component]}"
+          fi
+          unset map
```

Comment:
I wouldn't unset it. This is the end of the script and the pod is going to go away. Also, with the current code, any failure in a skopeo task will prevent this line to be executed. If you really want to run it, you should use a trap.